metadetect/base.py

```diff
@@ -0,0 +1,414 @@
+import logging
+import numpy as np
+import ngmix
+from ngmix.gexceptions import BootPSFFailure
+import esutil as eu
+from . import detect
+from . import fitting
+from . import procflags
+from . import shearpos
+from .mfrac import measure_mfrac
+
+logger = logging.getLogger(__name__)
+
+
+class BaseMetadetect(dict):
```

Comment:
I've separated out the Metadetect classes using sep from the LSST ones. The dev of these two shouldn't be coupled. Thus I copied the old Metadetect class into a new thing called BaseMetadetect.

Reply:
have you verified the lsst stuff still works?

Reply:
No I have not. I don't have a great local setup right now. We can/should probably add a test for this if we can. Would you mind if we punted to another PR for that test?